### PR TITLE
REST API: Add Header-Only parameter to blocks endpoint, return error if there are too many transactions.

### PR DESCRIPTION
**Pull request by shiqizng**
## Summary

#332 Adding header-only parameter to GET /v2/blocks/{round-number} which would return a block excluding the transactions. If the number of the transactions in a request block exceeds MaxTransactionsLimit, the endpoint return err code 400 when header-only parameter isn't set. 

## Test Plan
new unit tests and integration tests.

### CODE REVIEW

**Review by AlgoStephenAkiki — APPROVED** [review]

File: idb/idb.go [hunk 1]
```diff
@@ -401,3 +408,11 @@ type Health struct {
 type NetworkState struct {
 	GenesisHash crypto.Digest `codec:"genesis-hash"`
 }
+
+// MaxTransactionsError records the error when transaction counts exceeds MaxTransactionsLimit.
+type MaxTransactionsError struct {
+}
+
+func (e MaxTransactionsError) Error() string {
+	return "Max transactions limit exceeded. header-only flag should be enabled"
```
Comment (winder) [inline idb/idb.go:417]: Could you put this string in `error_messages.go`?

File: idb/postgres/postgres_integration_test.go [hunk 2]
```diff
@@ -375,8 +375,7 @@ func TestBlockWithTransactions(t *testing.T) {
 	//////////
 	// When // We call GetBlock and Transactions
 	//////////
-	_, txnRows0, err := db.GetBlock(
-		context.Background(), round, idb.GetBlockOptions{Transactions: true})
+	_, txnRows0, err := db.GetBlock(context.Background(), round, idb.GetBlockOptions{Transactions: true, MaxTransactionsLimit: 100})
```
Comment (winder) [inline idb/postgres/postgres_integration_test.go:378]: Could you add another test which triggers the MaxTransactionError case?

File: idb/idb.go (hunk 1 above)
Comment (shiqizng) [reply idb/idb.go]: added.

File: idb/postgres/postgres.go [hunk 3]
```diff
@@ -480,6 +481,9 @@ func (db *IndexerDb) GetBlock(ctx context.Context, round uint64, options idb.Get
 		results := make([]idb.TxnRow, 0)
 		for txrow := range out {
 			results = append(results, txrow)
+			if uint64(len(results)) > options.MaxTransactionsLimit {
```
[the remark this answers is not in the document]
Comment (shiqizng) [reply idb/postgres/postgres.go:484]: I think with the limit filter added, we're ok with `out` channel. When line 484 is true, `out` would be empty.

[the remark this answers is not in the document]
Comment (winder) [reply idb/postgres/postgres.go:484]: I think you're right, but the return inside the range seems dangerous. Here's an example of clearing out the channel (I thought there were others but this is the only one I could find): https://github.com/algorand/indexer/blob/develop/api/handlers.go#L1216

Could you move the if statement to the outside of the range statement? It should still be efficient, thanks to the limit, but would also ensure that the channel is closed.

**Review by winder — APPROVED** [review]

**Pull request merged**